modules/xfem/src/base/XFEM.C

```diff
@@ -1507,3 +1557,162 @@ XFEM::getXFEMqRuleOnSurface(std::vector<Point> & intersection_points, std::vecto
     }
   }
 }
+
+void
+XFEM::saveSolutionForNode(const Node * node_to_store_to,
+                          const Node * node_to_store_from,
+                          SystemBase & sys,
+                          std::map<unique_id_type, std::vector<Real>> & stored_solution,
+                          const NumericVector<Number> & current_solution,
+                          const NumericVector<Number> & old_solution)
+{
+  const std::vector<MooseVariable *> & vars = sys.getVariables(0);
+  _stored_solution_scratch.clear();
```

Comment:
So why is the scratch vector a class member instead of a local member? You are reallocating and copying it around anyway. In fact, you would be better off with using `::assign` here to avoid reallocation here if that's the optimization you are after.